js/background_scripts/background.js

```diff
@@ -74,43 +74,79 @@ chrome.runtime.onMessage.addListener(
 //Setup Icon Paths
 var notCompatibleIconPath = chrome.runtime.getURL("images/iconNotCompatible.png");
 var CompatibleIconPath = chrome.runtime.getURL("images/icon.png");
+var allTabs = {}
 
-//Check if the newly Selected Tab is playable
+//Check if the newly Selected Tab is compatible
 chrome.tabs.onActivated.addListener(function(info){
-    chrome.tabs.getSelected(null,function(tab){
-        var tabURL = tab.url;
+    chrome.tabs.query({active: true,lastFocusedWindow: true}, function (tab) {
+        var tabURL = tab[0].url;
+        var tabID = tab[0].id;
+        
+        //Check if Tab has the required infos
+        if(allTabs[tabID] == null){
+            allTabs[tabID] = {needCheck: true, isCompatible: false};
+        }
 
-        for (var i = 0; i < allModules.length; i++) {
-            var module = allModules[i];
-            if (module.canHandleUrl(tabURL) && !isContextMenuCreated(tabURL)){
+        //Check if Tab needs to be Checked
+        if(!allTabs[tabID].needCheck){
+            if(allTabs[tabID].isCompatible)
                 chrome.browserAction.setIcon({path: CompatibleIconPath});
-                break;
-            }
-            //Reduce Flickering of Icon
-            if(i == allModules.length-1)
+            else
                 chrome.browserAction.setIcon({path: notCompatibleIconPath});
+            return;
         }
+
+        //Check if Tab is Compatible
+        checkIfTabIsCompatible(tabURL,tabID);
     })
 });
 
-//Recheck the selected Tab when any Tab updates
+//Set needCheck flag to true when Tab Updates and recheck currently selected Tab
 chrome.tabs.onUpdated.addListener(function(tabID, changes){
```

Comment:
This event is fired when a Tab is updated. We get the updated tab ID.
Could you please explain why you query the tabs to get the current one and update it?
Isn't it redundant?

Reply:
The currently selected Tab gets rechecked, because when you press a link it only sets the needCheck flag to true, it doesn´t actually recheck the updated tab.

Will try to change it today though.